website/docs/guides/customize-schema-alias.md

```diff
@@ -0,0 +1,486 @@
+ 
+
+    - Note that for automated CI jobs, the schema name is derived from the job number and PR number: `dbt_cloud_pr_<job_id>_<pr_id>`
+    - If a custom schema has been defined, it will concatenate the schema from above with the custom one.
+    - For example, if the configured schema is `dbt_myschema` and the custom one is `marketing`, the objects will be created under `dbt_myschema_marketing`.
+
+- The object name depends on whether an [alias](/reference/resource-configs/alias) has been defined on the model:
+    - If no alias is defined, the object will be created with the same name as the model, without the `.sql` or `.py` at the end.
+    - If an alias is defined, the object will be created with the configured alias.
+
+Those default rules are a great starting point, and many people stick with those without customizing them.
+
+It allows developers to work in their own test schemas (sandboxes) and makes it possible to have different feature requests updating the same models without conflicting with one another.
+
+
+## How to customize this behavior
+
+While the default behavior would fit the needs of most dbt users, there are occasions where the approach doesn’t fit with the existing data warehouse governance and design.
+
+For example, dbt expects that it has the rights to create a schema on the fly (and we recommend that the users running dbt have this ability), but it might not be allowed at your company.
+
+Or, you might have many custom schemas and want to avoid having your dev database showing the combination of all developer schemas and custom schemas.
+
+Alternatively, you may even want to create models in schemas based on the feature branch name in development rather than the developer name.
+
+For this reason, dbt offers three macros to customize what objects are created in the Data Warehouse:
+
+- [`generate_database_name()`](/docs/build/custom-databases#generate_database_name)
+- [`generate_schema_name()`](/docs/build/custom-schemas#how-does-dbt-generate-a-models-schema-name)
+- [`generate_alias_name()`](/docs/build/custom-aliases#generate_alias_name)
+
+By overwriting one or multiple of those macros, we can tailor where dbt objects are created in the data warehouse and align with any existing requirement.
+
+:::note Key concept
+
+The key concept is that models run from two different contexts need to result in different objects in the data warehouse to avoid people overriding each others models. For example running by developer 1 or by developer 2; or running by developer 1 and running in production.
+
+:::
+
+
+When those macros are customized, we often leverage some of the following logic:
+
+In dbt Cloud, we recommend environment variables as they can be set at the environment and all jobs will automatically inherit those values.
```

Comment:
- In dbt Cloud, we recommend utilizing [environment variables](/docs/build/environment-variables) to define where the dbt invocation is occurring (dev/stg/prod). They can be set at the environment level, and all jobs will automatically inherit the default values. We'll add jinja logic (`if/else/endif`) to identify whether the run happens in dev, prod, Ci, etc.